test/testdata/parser/error_recovery/unmatched_block_args_2.rb

```diff
@@ -4,7 +4,7 @@ module Opus::Log
   def self.info; end
 end
 
-class A # error: class definition in method body
```

Comment:
We now only raise this error if we are in a method definition. Lack of this error triggered some changes on the desugar file.